### DISCUSSION TITLE
non-english model error Can't find model 'corpus/de_vectors' with nlp.initialize

### DISCUSSION THREAD

**Discussion by smartrestart** [explosion/spaCy#7202] [Help: Coding & Implementations]
## How to reproduce the behaviour
Hi, I have an issue that has two sides of it, I think.
Side 1:

With Spacy 3.0 I tried to install a German model with:

python -m spacy download de_core_news_lg

and also already tried with:

pip install https://github.com/explosion/spacy-models/releases/download/de_core_news_lg-3.0.0/de_core_news_lg-3.0.0.tar.gz

I tried to simplify my code so that you can replicate the issue with the following:

```
import spacy
#import de_core_news_lg
nlp = spacy.load('de_core_news_lg')
nlp.initialize()
```

The import de_core_news_lg I also tried to include, as I saw it somewhere else. But also this didn't work.

The error I get is:

```
Traceback (most recent call last):
  File "error_test.py", line 5, in <module>
    nlp.initialize()
  File "/home/z01/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/language.py", line 1225, in initialize
    init_vocab(
  File "/home/z01/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/training/initialize.py", line 112, in init_vocab
    load_vectors_into_model(nlp, vectors)
  File "/home/z01/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/training/initialize.py", line 122, in load_vectors_into_model
    vectors_nlp = load_model(name)
  File "/home/z01/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/util.py", line 329, in load_model
    raise IOError(Errors.E050.format(name=name))
OSError: [E050] Can't find model 'corpus/de_vectors'. It doesn't seem to be a Python package or a valid path to a data directory.

```
I even tried to install the model outside of the virtual environment. I also tried to run it as .py file as also in jupyter notebook.

Side 2: 
Originally, this was trying to do something else that also led an issue. I am sending it here as one bug report as I assume it could have the same reason.
Actually I didn't want to do nlp.initialize but nlp.resume_training() with the following code that actually fails when trying to nlp.update.

```
# add NER to the pipeline and the new label
import de_core_news_lg
nlp = spacy.load('de_core_news_lg')
ner = nlp.get_pipe("ner")
ner.add_label("LOCALITY")
from spacy.training import Example
from spacy.tokens import Doc


# get the names of the components we want to disable during training
pipe_exceptions = ["ner", "trf_wordpiecer", "trf_tok2vec"]
other_pipes = [pipe for pipe in nlp.pipe_names if pipe not in pipe_exceptions]

# start the training loop, only training NER
epochs = 30
optimizer = nlp.resume_training()
#optimizer = nlp.initialize()
with nlp.disable_pipes(*other_pipes), warnings.catch_warnings():
    warnings.filterwarnings("once", category=UserWarning, module='spacy')
    sizes = compounding(1.0, 4.0, 1.001)
    
    # batch up the examples using spaCy's minibatc
    for epoch in range(epochs):
        #examples = TRAIN_DATA
        random.shuffle(TRAIN_DATA)
        #examples = []
        #for text, annots in TRAIN_DATA:    
        #    examples.append(Example.from_dict(nlp.make_doc(text), annots))
        #nlp.initialize(lambda: examples)
        
        #example = Example.from_dict(dictionary)
        #example = Example(doc, annotations)
        losses = {}
        
        for batch in minibatch(TRAIN_DATA, size=8):
            for text, annotations in batch:
                doc = nlp(text)
                example = Example.from_dict(doc, annotations)
            #    print(example)
                nlp.update([example])#, drop=0.35, sgd=optimizer, losses=losses)
            
        #nlp.update([example], sgd=optimizer, drop=0.35, losses=losses)

        print("Losses ({}/{})".format(epoch + 1, epochs), losses)
```
Here the error message is:
```
alueError                                Traceback (most recent call last)
<ipython-input-19-407d57cee8ab> in <module>
     38                 example = Example.from_dict(doc, annotations)
     39             #    print(example)
---> 40                 nlp.update([example])#, drop=0.35, sgd=optimizer, losses=losses)
     41 
     42         #nlp.update([example], sgd=optimizer, drop=0.35, losses=losses)

~/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/language.py in update(self, examples, _, drop, sgd, losses, component_cfg, exclude)
   1104             if name in exclude or not hasattr(proc, "update"):
   1105                 continue
-> 1106             proc.update(examples, sgd=None, losses=losses, **component_cfg[name])
   1107         if sgd not in (None, False):
   1108             for name, proc in self.pipeline:

~/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/pipeline/transition_parser.pyx in spacy.pipeline.transition_parser.Parser.update()

~/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/pipeline/transition_parser.pyx in spacy.pipeline.transition_parser.Parser.get_batch_loss()

~/nlp_learn/statbot/.statbot/lib/python3.8/site-packages/spacy/pipeline/_parser_internals/ner.pyx in spacy.pipeline._parser_internals.ner.BiluoPushDown.set_costs()

ValueError: 
```
Yes: ValueError is empty??

## Info about spaCy

- **spaCy version:** 3.0.0
- **Platform:** Linux-5.8.0-43-generic-x86_64-with-glibc2.29
- **Python version:** 3.8.5
- **Pipelines:** de_core_news_sm (3.0.0), de_core_news_lg (3.0.0), de_dep_news_trf (3.0.0)

* Operating System: Ubuntu 20.4
* Python Version Used: 3.8.5
* spaCy Version Used: 3.0.0
* Environment Information:

**Comment by svlandeg** [comment]
Hi! This type of question is very suited for the new discussions forum, so I'll move it there. You'll get a notification that this was closed, but the thread remains open on the forum and you 'll be forwarded there.

**Comment by svlandeg** [comment]
```
import spacy
nlp = spacy.load('de_core_news_lg')
nlp.initialize()
```

As you're loading in an already initialized and trained pipeline, you shouldn't call `nlp.initialize`, because that would basically wipe all weights from the models. Additionally, the pipeline tries to find the initial vectors again, but those aren't present on your system (and don't have to be). See also a more detailed explanation on a similar issue [here](https://github.com/explosion/spaCy/discussions/6972#discussioncomment-350499).

The error message is confusing though, and PR https://github.com/explosion/spaCy/pull/7104 will make it more clear what the problem is here.

**Reply by ruizcrp** [reply]
Hi, Thank you for your answer. OK so the nlp.initialize part does not make sense. What about the second part where nlp.update gives a valuerror that is empty? Or am I doing something wrong?

**Reply by ruizcrp** [reply]
Here I also give the output of the print(example) command:

{'doc_annotation': {'cats': {}, 'entities': ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'B-LOCALITY', 'L-LOCALITY', 'O', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Wie', 'viel', 'Bevölkerung', ':', 'reformiert', ' ', 'hat', ' ', 'auf', 'Gemeindeebene', ' ', '?'], 'SPACY': [True, True, False, True, True, False, True, False, True, True, False, False], 'TAG': ['', '', '', '', '', '', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], 'DEP': ['', '', '', '', '', '', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}}
{'doc_annotation': {'cats': {}, 'entities': ['O', 'O', 'O', 'O', 'O', 'O', 'B-LOCALITY', 'L-LOCALITY', 'O', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Wie', 'viel', 'Zahnärzte', ' ', 'hat', ' ', 'pro', 'Bezirk', ' ', '?'], 'SPACY': [True, True, True, False, True, False, True, True, False, False], 'TAG': ['', '', '', '', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9], 'DEP': ['', '', '', '', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0, 0, 0, 0]}}
{'doc_annotation': {'cats': {}, 'entities': ['O', 'O', 'O', 'O', 'O', 'U-LOC', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Diskutieren', 'Sie', 'über', 'diesen', 'Artikel', 'Forum', '-'], 'SPACY': [True, True, True, True, True, True, False], 'TAG': ['', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6], 'DEP': ['', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0]}}
{'doc_annotation': {'cats': {}, 'entities': ['O', 'O', 'O', 'O', 'O', 'U-MISC', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Diese', 'müssen', 'gerettet', 'und', 'der', 'Brand', 'gelöscht', 'werden', '“', ',', 'so', 'der', 'Wehrleiter', '.'], 'SPACY': [True, True, True, True, True, True, True, False, False, True, True, True, False, False], 'TAG': ['', '', '', '', '', '', '', '', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', '', '', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', '', '', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', '', '', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13], 'DEP': ['', '', '', '', '', '', '', '', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}}
{'doc_annotation': {'cats': {}, 'entities': ['U-LOC', 'O', 'O', 'O', 'O', 'O', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Hitler-Parodien', 'gab', 'es', 'so', 'einige', 'seither', '.'], 'SPACY': [True, True, True, True, True, False, False], 'TAG': ['', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6], 'DEP': ['', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0]}}
{'doc_annotation': {'cats': {}, 'entities': ['O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O', 'B-LOCALITY', 'L-LOCALITY', 'O'], 'links': {}}, 'token_annotation': {'ORTH': ['Was', 'ist', 'der', 'Anteil', 'an', 'KRW', 'Wähleranteil', 'GLP', '  ', 'auf', 'Bezirksebene', '?'], 'SPACY': [True, True, True, True, True, True, True, True, False, True, True, False], 'TAG': ['', '', '', '', '', '', '', '', '', '', '', ''], 'LEMMA': ['', '', '', '', '', '', '', '', '', '', '', ''], 'POS': ['', '', '', '', '', '', '', '', '', '', '', ''], 'MORPH': ['', '', '', '', '', '', '', '', '', '', '', ''], 'HEAD': [0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11], 'DEP': ['', '', '', '', '', '', '', '', '', '', '', ''], 'SENT_START': [1, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0]}}

So "example" itself is not empty.

**Reply by ruizcrp** [reply]
I've also read that the length of train_data could be an issue: It contains now over 5900 elements and I get the same issue.

**Reply by adrianeboyd** [reply]
The `set_costs` `ValueError` is a problem with really small batches / toy examples that we should look into, so we can provide a better error message at the very least.

The problem in the script above is that you're updating with individual texts rather than larger batches. Create a list of `examples` with more texts in one batch and you should be able to train the model.

In general, we'd recommend using `spacy train` instead of this sort of script because it handles a lot of these details for you and comes with better default setting for batching, etc.
